### ISSUE TITLE
Changes to .js files are no longer picked up by watchify

### ISSUE THREAD

**Issue by motherhubbard** [jmreidy/grunt-browserify#343]
OSX Yosemite using "grunt-browserify": "3.8.0".

Grunt config:
      app: {
        options: {
          watch: true,
          browserifyOptions: {
            paths: ['./node_modules', '<%= srcDir %>']
          },
          external: browserifyLibs.concat('common')
        }
      }

This was working until we did an new npm install which bought in version 3.3.2 of watchify. Ive noticed this no longer uses chokidar suggesting there have been changing the the file watching area.

npm installing watchify@3.2.2 resolves this issue.

**Comment by tleunen** [comment]
`grunt-browserify` is now at `4.0.0`. So unless there's an issue with the latest version, I'll close this

**Comment by motherhubbard** [comment]
FWIW I'd expect it to be in 4.0.0 also as it uses "watchify": "^3.3.1".

**Comment by tleunen** [comment]
Watchify 3.3.1 still uses chokidar.
Is it working with the CLI?

**Comment by Myztiq** [comment]
The problem is watchify added this commit: https://github.com/substack/watchify/commit/db27889c699a947b477a19313a862a9cee6be772

which bumped browserify major version 10.x to 11.x and then promptly made a new version 3.3.0 from 3.2.0 bumping minor versions.

This broke watchify for everyone who was expecting browserify 10.x

**Comment by Myztiq** [comment]
An easy fix? Lock versions of submodules until we can trust them to not make minor releases for major changes.

**Comment by motherhubbard** [comment]
That sounds good. Fwiw we simply included 3.2.2 in our package json. This worked for 3.8 but probably wouldn't work with the requirement above for v4

**Comment by motherhubbard** [comment]
Although installing 3.3.1 may have done if it still uses chokidar ;-)

**Comment by tleunen** [comment]
So, is there a real issue here?
Still not sure to understand. Watchify uses Browserify 11. We are using Browserify 11. So...

**Comment by Myztiq** [comment]
@tleunen 3.8.0 uses browserify 10, and it uses a version of watchify that uses browserify 11. That's the mismatch that's causing failure.

**Comment by tleunen** [comment]
oh I see. Then upgrading to grunt-browserify 4.0 should resolve it?

**Comment by Myztiq** [comment]
Correct, but that also means everyone needs to upgrade to browserify 11, which may have other unintended consequences, for instance it broke a few things in our build which we managed to solve, just a little more effort and validation is required for that upgrade. Ideally there would be a `3.8.1` release that makes the dependencies fixed. I mean, ideally locking the submodules would be in 4.0 as well...

**Comment by tleunen** [comment]
Watchify seems to make a minor update when browserify upgrades. So we might solve this by setting the dependency of watchify with a tilde. What do you think?

**Comment by Myztiq** [comment]
Yea, that makes sense to me!
